rio_tiler/io/stac.py

```diff
@@ -185,7 +185,7 @@ def _get_href(self, assets: Sequence[str]) -> Sequence[str]:
         """Validate asset names and return asset's url."""
         for asset in assets:
             if asset not in self.assets:
-                raise InvalidBandName(f"{asset} is not a valid asset name.")
+                raise InvalidAssetName(f"{asset} is not a valid asset name.")
```

Comment:
I think we can prevent looping through the assets twice.  The result set will also contain all assets which are invalid.

```python
def _get_href(self, assets: Sequence[str]) -> Sequence[str]:
    invalid_assets = set(assets) - set(self.assets)
    if invalid_assets:
        raise InvalidAssetName(f"{invalid_assets} is not a valid asset name")
    return [self.item["assets"][asset]["href"] for asset in assets]
```